src/store/fs.rs

```diff
@@ -644,25 +687,24 @@ trait HashSpecificCommand: HashSpecific + Send + 'static {
     ) where
         Self: Sized,
     {
+        let span = tracing::Span::current();
         let task = manager
-            .spawn_boxed(
-                self.hash(),
-                Box::new(|x| {
-                    Box::pin(async move {
-                        match x {
-                            SpawnArg::Active(state) => {
-                                self.handle(state).await;
-                            }
-                            SpawnArg::Busy => {
-                                self.on_error().await;
-                            }
-                            SpawnArg::Dead => {
-                                self.on_error().await;
-                            }
+            .spawn(self.hash(), |arg| {
+                async move {
+                    match arg {
+                        SpawnArg::Active(state) => {
+                            self.handle(state).await;
                         }
-                    })
-                }),
-            )
+                        SpawnArg::Busy => {
+                            self.on_error(arg).await;
+                        }
+                        SpawnArg::Dead => {
+                            self.on_error(arg).await;
+                        }
+                    }
+                }
+                .instrument(span)
+            })
             .await;
         if let Some(task) = task {
             tasks.spawn(task);
```

Comment:
IMHO it would make more sense to `.instrument` here instead of above. I don't think there currently are, but in theory there could be logs between the callback to `spawn` and the future returned from `spawn` that wouldn't be instrumented properly otherwise.